src/substrait/textplan/tests/ParseResultMatchers.cpp

```diff
@@ -0,0 +1,153 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/textplan/tests/ParseResultMatchers.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include "substrait/textplan/ParseResult.h"
+#include "substrait/textplan/SymbolTable.h"
+#include "substrait/textplan/SymbolTablePrinter.h"
+
+namespace io::substrait::textplan {
+namespace {
+
+std::vector<std::string> symbolNames(
+    const std::vector<std::shared_ptr<SymbolInfo>>& symbols) {
+  std::vector<std::string> names;
+  for (const auto& symbol : symbols) {
+    names.push_back(symbol->name);
+  }
+  return names;
+}
+
+} // namespace
+
+class ParsesSuccessfullyMatcher {
+ public:
+  using is_gtest_matcher = void;
+
+  explicit ParsesSuccessfullyMatcher() = default;
+
+  static bool MatchAndExplain(
+      const ParseResult& result,
+      std::ostream* /* listener */) {
+    return result.successful();
```

Comment:
By default it's going to output a full summary of the ParseResult including the error message.  It could be better and show me specific differences (perhaps a typo in a single message) but it may also end up being more verbose.  Something to fiddle with over time.

.../substrait-cpp/src/substrait/textplan/converter/tests/BinaryToTextPlanConversionTest.cpp:136: Failure
Value of: result
Expected: parses successfully
  Actual: {
  "1:10 → Message type "substrait.proto.Plan" has no field named "gibberish".",
} (of type io::substrait::textplan::ParseResult)